### PR TITLE
Add, Subtract, Multiply Many

### LINKED ISSUE #74: Add/Subtract multiple amounts together
It would be nice if the `Money` struct has methods to add or subtract multiple values. For example:

````go
func( *m Money) AddMultiple(om... *Money) (*Money, error){

}

func( *m Money) SubtractMultiple(om... *Money) (*Money, error){

}
````

I write a lot of test cases that calculates money and invoices and I often need to add up multiple values in a test case to check if the calculated amount is correct. It's possible to do this using the `Add()` method, but it's quite tedious to have to check the error for each addition. I am currently using a helper function to do this, but I think this would be a very useful addition to the library.

### PR DESCRIPTION
**Pull request by AyoobMH**
* **Please check if the PR fulfills these requirements**
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
 Addition, subtraction, and multiplication can only be done with single values.

* **What is the new behavior (if this is a feature change)?**
 Addition, subtraction, and multiplication using many values.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
NO

* **Other information**:
 * Fixes #74 
 * Credit to @SepehrAsh for opening #78 ,but it got closed.

### CODE REVIEW

**Comment by Rhymond** [comment]
Hi @AyoobMH 

Thanks for the PR! I was just thinking, maybe we can update original methods to accept variadic parameters instead?
```
func (m *Money) Add(om ...*Money) (*Money, error)
```

In that way the functionality to add multiple would exists and it would not introduce any breaking changes. Of course, I would expect to do a sanity check `if len(om) != 0` and return error before applying operation.

What do you think?

**Comment by AyoobMH** [comment]
Hi @Rhymond, thanks for your efforts.
I think updating original methods to have variadic parameters is even better, it wouldn't introduce any redundant code. Are you thinking of updating only `Add`, `Subtract`, `Multiply`? I would be happy to open another PR for that.

**Comment by Rhymond** [comment]
> Hi @Rhymond, thanks for your efforts.
> 
> I think updating original methods to have variadic parameters is even better, it wouldn't introduce any redundant code. Are you thinking of updating only `Add`, `Subtract`, `Multiply`? I would be happy to open another PR for that.

Exactly! Yes `Add`, `Subtract` and `Multiply`. Appreciate your time 🙏

**Comment by AyoobMH** [comment]
Hi @Rhymond 

I have modified the PR and added variadic parameters. For checking parameters, I have used panic instead of an error, I believe not passing any parameter is considered a bad use of the package and should not happen during runtime. What do you think?

File: money.go [hunk 1]
```diff
@@ -198,26 +198,56 @@ func (m *Money) Negative() *Money {
 }
 
 // Add returns new Money struct with value representing sum of Self and Other Money.
-func (m *Money) Add(om *Money) (*Money, error) {
-	if err := m.assertSameCurrency(om); err != nil {
-		return nil, err
+func (m *Money) Add(om ...*Money) (*Money, error) {
+	if len(om) == 0 {
+		panic("At least one Money is required to add")
```
Comment (totemcaf) [inline money.go:203]: I think you can return the original value instead of panic because it is not incorrect mathematically.

File: money.go [hunk 2]
```diff
@@ -198,26 +198,56 @@ func (m *Money) Negative() *Money {
...
+	k := New(0, m.currency.Code)
+
+	for i := 0; i < len(om); i++ {
+		if err := m.assertSameCurrency(om[i]); err != nil {
+			return nil, err
+		}
+
+		k.amount = mutate.calc.add(k.amount, om[i].amount)
+	}
+
```
Comment (totemcaf) [inline money.go:215]: You can use the alternative form:
```suggestion
	for _, m2:= range om {
		if err := m.assertSameCurrency(m2); err != nil {
			return nil, err
		}

		k.amount = mutate.calc.add(k.amount, m2.amount)
	}

```

File: money.go [hunk 3]
```diff
@@ -198,26 +198,56 @@ func (m *Money) Negative() *Money {
 }
 
 // Add returns new Money struct with value representing sum of Self and Other Money.
-func (m *Money) Add(om *Money) (*Money, error) {
-	if err := m.assertSameCurrency(om); err != nil {
-		return nil, err
+func (m *Money) Add(om ...*Money) (*Money, error) {
```
Comment (totemcaf) [inline money.go:201]: It's typically to add an "s" to the name to denote an array of:
```suggestion
func (m *Money) Add(ms ...*Money) (*Money, error) {
```

File: money.go [hunk 4]
```diff
@@ -198,26 +198,56 @@ func (m *Money) Negative() *Money {
...
+	return &Money{amount: mutate.calc.add(m.amount, k.amount), currency: m.currency}, nil
 }
 
 // Subtract returns new Money struct with value representing difference of Self and Other Money.
-func (m *Money) Subtract(om *Money) (*Money, error) {
-	if err := m.assertSameCurrency(om); err != nil {
-		return nil, err
+func (m *Money) Subtract(om ...*Money) (*Money, error) {
+	if len(om) == 0 {
+		panic("At least one Money is required to subtract")
```
Comment (totemcaf) [inline money.go:222]: Same for add, you can return original value instead of panicking.

File: money.go [hunk 5]
```diff
@@ -198,26 +198,56 @@ func (m *Money) Negative() *Money {
...
+
+	return &Money{amount: mutate.calc.subtract(m.amount, k.amount), currency: m.currency}, nil
 }
 
 // Multiply returns new Money struct with value representing Self multiplied value by multiplier.
-func (m *Money) Multiply(mul int64) *Money {
-	return &Money{amount: mutate.calc.multiply(m.amount, mul), currency: m.currency}
+func (m *Money) Multiply(mul ...int64) *Money {
+	if len(mul) == 0 {
+		panic("At least one multiplier is required to multiply")
```
Comment (totemcaf) [inline money.go:241]: Here is more controversial to return the original value, so I think it is ok to panic

**Review by totemcaf — CHANGES_REQUESTED** [review]

**Comment by AyoobMH** [comment]
@totemcaf Thanks for the review, the requested changes were made!

**Comment by AyoobMH** [comment]
Hi @Rhymond @totemcaf , this PR is ready for review, would appreciate your review, thanks in advance!

**Comment by AyoobMH** [comment]
Hi @Rhymond, any updated for this PR, thanks in advance!

**Review by Rhymond — APPROVED** [review]
Sorry for late response, LGTM

**Pull request merged**